shm/Map.cpp

```diff
@@ -1,6 +1,30 @@
 #include "Map.hpp"
-
 #include <algorithm>
+#include <random>
+#include <iostream>
+
+Map::Map(){
+    islandsLocations_.reserve(amountIslands);
+    generateIslands();
+}
+
+void Map::generateIslands() {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib(widthIsland, heightIsland);
+    
+      for (size_t i = 0; i < amountIslands; i++) { 
+        size_t positionX; 
+        size_t positionY; 
+        do { 
+            size_t positionX = distrib(gen); 
+            size_t positionY = distrib(gen);
+        } while (std::any_of(islandsLocations_.begin(), islandsLocations_.end(), [positionX, positionY](auto& island) { 
+            return island.getPosition() == Coordinates(positionX, positionY); 
```

Comment:
I am not sure from which scope `positionX` and `positionY` come? Are those from lines 20 and 21 (from inside do-while loop) or from lines 17 and 18 (from inside for loop)?

But for me it looks like  `positionX` and `positionY`  from inside do-while loop (these from lines 20 and 21) are never used. Try to comment those for a moment and check if while the IDE will warn you or not. Later, comment lines 17 and 18 and then check again.

This is related to my earlier comment that you re-declared variables.

Reply:
in Line 20 and 21 X and Y values are being generated. But she unnecessarily declares them every loop as pointed in other comment.